lib/ValidationError.test.ts

```diff
@@ -0,0 +1,52 @@
+import * as zod from 'zod';
+
+import { ValidationError } from './ValidationError';
+
+describe('ValidationError', () => {
+  describe('constructor', () => {
+    test('accepts message', () => {
+      const message = 'Invalid email coyote@acme';
+
+      const err = new ValidationError(message);
+      expect(err.message).toBe(message);
+      expect(err.details).toEqual([]);
+    });
+
+    test('accepts message with cause', () => {
+      const message = 'Invalid email coyote@acme';
+
+      const err = new ValidationError(message, {
+        cause: new Error('foobar'),
+      });
+      expect(err.message).toBe(message);
+      expect(err.details).toEqual([]);
```

Comment:
[question] Do we want to extend the assertions with an extra one that ensures `err.cause` is the same as the one we provided?

```ts
expect(err.cause).toBe(inputError)
```

Reply:
Sure, valid suggestion.